### ISSUE TITLE
Users getting session expired right after sign in

### ISSUE THREAD

**Issue by brunovsiqueira** [aws-amplify/amplify-flutter#1271]
Hi there.

We have two android users in production which are not being able to use our app because Amplify returns session expired right after signIn  when we call `fetchAuthSession`.

Inspecting this user's logs, what I see is the following:

1- We check user session by calling (`Amplify.Auth.getCurrentUser()`, `Amplify.Auth.fetchAuthSession`) when the app is opened. We consider session expired if `InvalidStateException` or `SessionExpiredException` is thrown and then we call `Amplify.Auth.signOut()` if session has expired. In the mentioned case, this user session is considered expired and he is logged out.

2- After that, the user called signIn successfully and "isSignedIn" flag returned true.

3- Right after signIn (sequentially), `Amplify.Auth.fetchAuthSession` is called in order to get user token (JWT) to call one of our APIs. When fetchAuthSession is called, this is the result according to the logs: 
`(message: Your session has expired., recoverySuggestion: Please sign in and reattempt the operation., underlyingException: null)`
Because user session has expired right after signIn, the user is not able to get the token and consequently is not able to consume none of our APIs. 

This behavior is completely weird since the user has just logged in and is getting session expired. I was not able to reproduce it. 

Since we have production users facing this problem, can anyone please help me understand/solve the problem?

**Comment by brunovsiqueira** [comment]
@haverchuck @Amplifiyer

**Comment by fjnoyp** [comment]
Hi @brunovsiqueira 

Have some followup questions to better help you here: 

1) Is this Android only?
2) Are you able to reproduce this on your side when deploying your app to an Android device or emulator? 
3) Which Amplify CLI version are you using? 

4) We have a similar [issue](https://github.com/aws-amplify/amplify-flutter/issues/1204#issuecomment-1015010457), it's a bit long and we're still working on resolution on it.  For them, calling `await getCurrentUser()` right after `await signIn()` throws a `SignedOutException` - can you verify if you have the same behavior?

**Comment by brunovsiqueira** [comment]
Hi @fjnoyp 

1. Only android AFAIK. In my app logs I only see this problem for Android devices.
2. No, I wasn't.
3.  Amplify config:
``dart
String amplifyConfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "${getPoolIdForEnv()}",
                        "AppClientId": "${getAppClientIdForEnv()}",
                        "Region": "${getRegionForEnv()}"
                    }
                },
                "Auth": {
                    "Default": {
                        "authenticationFlowType": "USER_SRP_AUTH"
                    }
                }
               
            }
        }
    }
}''';
``dart
4. I tested here and the answer is no. getCurrentUser right after signIn works properly.

Also, my amplify versions:
```
amplify_auth_cognito: 0.2.10
amplify_core: 0.2.10
amplify_flutter: 0.2.10
```

**Comment by haverchuck** [comment]
@brunovsiqueira Do you know which devices these are occurring on?

**Comment by brunovsiqueira** [comment]
@haverchuck  Samsung SM-G973F (Galaxy S10), SM-N770F (Galaxy Note 20)

**Comment by dnys1** [comment]
Could this explain the issue you're seeing?

https://github.com/aws-amplify/amplify-android/issues/1485#issuecomment-915691808

**Comment by brunovsiqueira** [comment]
Hi @dnys1.

One of our users who reported the problem reinstalled the app from scratch. The behavior that we see in our logs is even stranger:

1- Because he reinstalled the App, he had no session so no SessionExpired was thrown.
2- He tried to sign in (in theory by the first time, since he reinstalled the app) and he signedIn with "success" (which is confirmed if we look at cognito console). By success I mean no exception was thrown when calling the signIn method.
<img width="1392" alt="Screen Shot 2022-01-19 at 18 54 52" src="https://user-images.githubusercontent.com/18200284/150220207-00c11ca6-6e51-4af2-a3ae-39db11130b12.png">
3- Right after signIn we call fetchAuthSession to get JWT token and it throw SessionExpiredException.


One thing that may be linked to the issue you mentioned is the fact that no exception is thrown when we call signIn, but instead it is thrown we try to fetch user session by calling fetchAuthSession (not sure if this behavior is expected, maybe yes).

I don't think the issue you mentioned explains our problem, since even reinstalling the app the user gets SessionExpired when fetching session right after a "successful" sign in.

**Comment by dnys1** [comment]
Hmm okay, I will look into it more tomorrow. Does the object returned from the signIn call provide any info?

I did try to reproduce this both in an emulator (Pixel) and a physical device (Galaxy S8) and for after the access token expired and after the access & refresh tokens expired. For all cases, I got the expected behavior. Unfortunately, I'm not sure what else to test for this.. these kinds of bugs are obviously very difficult to triage.

**Comment by brunovsiqueira** [comment]
Actually I don't have access to the signIn object in my logs.

I think I am going to try to create an emulador which has same configurations of the mentioned devices (S10 and Note 10). I agree this bugs are very hard to reproduce.

**Comment by brunovsiqueira** [comment]
@dnys1 were you able to figure something out?

**Comment by dnys1** [comment]
Unfortunately, I haven't @brunovsiqueira - I'm at a bit of a loss. Would you be willing to open a ticket with the amplify-android team with the same information? It's possible they may have some more insight as to what could be happening.

**Comment by bwhiteTP** [comment]
We are having this problem as well.

**Comment by brunovsiqueira** [comment]
@dnys1 yes, I can. But it is happening in the flutter plugin.

**Comment by fjnoyp** [comment]
Indeed. The Amplify Flutter plug-in's functionality is built on top of Amplify Android which might be the root source of the problem. Just to confirm, the issue you encounter does not happen on iOS?

**Comment by brunovsiqueira** [comment]
@fjnoyp I haven't seen any of this issue logs on iOS, so I assume it is not happening on iOS but I am not totally sure.

**Comment by brunovsiqueira** [comment]
> We are having this problem as well.

@bwhiteTP is this problem happening for you only on Android as well? Do you have any information about the devices where it is happening?

**Comment by bwhiteTP** [comment]
This does not happen to us on iOS, only Android. It also only seems to happen with sign in with email. It does't happen on sign up and it doesn't happen every time.
The way I've reproduced it:
1. Sign in with SSO
2. Intentionally Sign out
3. Sign up with email
4. Intentionally Sign out
5. Sign in with email - Get signed out immediately

After I sign in with email and get signed out a few more times it will just start working randomly. So the issue seems to fix itself after several tries.

**Comment by brunovsiqueira** [comment]
We don't use SSO and we only sign in our user using "username", not email.

@bwhiteTP do you know in which device the error is happening?

**Comment by brunovsiqueira** [comment]
Complete logs:
```
SessionExpiredException(message: Your session has expired., recoverySuggestion: Please sign in and reattempt the operation., underlyingException: null)
#0      AmplifyAuthCognitoMethodChannel.fetchAuthSession (package:amplify_auth_cognito\/method_channel_auth_cognito.dart:246)
<asynchronous suspension>
#1      AmplifyAuthCognito.fetchAuthSession (package:amplify_auth_cognito\/amplify_auth_cognito.dart:121)
<asynchronous suspension>
#2      SessionService._fetchSession (package:btg_auth_core\/src\/services\/session_service.dart:23)
<asynchronous suspension>
#3      SessionService.getAccessJWTToken (package:btg_auth_core\/src\/services\/session_service.dart:34)
<asynchronous suspension>
#4      BasicAPI.getHeaders (package:btg_auth_core\/src\/services\/api\/basic_api.dart:27)
<asynchronous suspension>
#5      BasicAPI.httpGet (package:btg_auth_core\/src\/services\/api\/basic_api.dart:136)
<asynchronous suspension>
#6      GkRemoteDatasourceImpl.getSelfUser (package:btg_auth_core\/src\/datasources\/gk_remote_datasource.dart:23)
<asynchronous suspension>
#7      UserServiceImpl.getSelfUser (package:btg_auth_core\/src\/services\/user_service_impl.dart:28)
<asynchronous suspension>
#8      UserViewModel.fetchCurrentUserData (package:btg_auth_core\/src\/viewmodels\/user_viewmodel.dart:33)
<asynchronous suspension>
#9      LoginViewModel.login (package:btg_login\/src\/viewmodels\/login_viewmodel.dart:218)
<asynchronous suspension>
#10     _LoginPasswordPageState.build.<anonymous closure> (package:btg_login\/src\/pages\/login_password_page.dart:72)
<asynchronous suspension>
```

**Comment by fjnoyp** [comment]
Hi @brunovsiqueira @bwhiteTP thanks for sharing that information.  Based on the information you've shared this sounds like an issue that is coming from the underlying Amplify Android library. 

Can you do what @dnys1 suggested and create an issue on the Amplify Android git repo? 

https://github.com/aws-amplify/amplify-android/issues

**Comment by brunovsiqueira** [comment]
I have created the issue on the Amplify Android git repo: https://github.com/aws-amplify/amplify-android/issues/1635.

**Comment by fjnoyp** [comment]
Great thank you.  We can monitor from there.

**Comment by dorontal** [comment]
@brunovsiqueira your issue seems very similar to one that I have raised here a while ago and one which I've had for months and on which I am still waiting: it could be related to, or the same issue as this https://github.com/aws-amplify/amplify-flutter/issues/1204

**Comment by brunovsiqueira** [comment]
Just an update: it's happening in both iOS and Android. Also, it is happening a lot. Seems that even reinstalling the app the SessionExpiredException keep being thrown right after signIn.
This issue is highly critical and I think it must be given the properly urgency.

**Comment by brunovsiqueira** [comment]
I think the issue my users are having is the same. Seems that it happens in both login/relogin and also in both platforms (Android and iOS).
By looking at out logs, it is happening with several users.
These are the exceptions I see this users are getting:

```
SessionExpiredException(message: Session expired could not fetch user sub, recoverySuggestion: Invoke Auth.signIn to re-authenticate the user, underlyingException: null)

SignedOutException(message: There is no user signed in to retreive user sub, recoverySuggestion: Call Auth.signIn to sign in a user and then call Auth.fetchSession, underlyingException: null)
```

**Comment by brunovsiqueira** [comment]
Looking at an iOS user specific behavior:
1- User is logging in normally and using our app (I can see by his logs)
2- Following method is called and returns true because SessionExpiredException is thrown:

```
Future<bool> isSessionExpired() async {
    try {
      await _fetchSession();
      return false;
    } on SignedOutException {
      return false;
    } on SessionExpiredException {
      return true;
    } on InvalidStateException {
      return true;
    } on AuthException {
      rethrow;
    }
  }
```
3- By determining that session is expired, we call: "await Amplify.Auth.signOut()" in order to try to signout an user which his session expired.
4- After it user signs in "normally" (without throwing an exception) by entering his username and password and the app calls await Amplify.Auth.signIn. We check isUserSignedIn flag and it also returns true.
5- Right after it, we call "await Amplify.Auth.fetchAuthSession" and it throws SignedOutException. By throwing this exception, we are unable to get the access token.

**Comment by brunovsiqueira** [comment]
In android, the behavior changes depending on the user. One thing never changes: user gets SessionExpiredException right after signIn (on iOS seems that he get SignedOutException).
We have some users who got SessionExpiredException even in their first signIn attempts which is really sad.

**Comment by brunovsiqueira** [comment]
Digging deep into the problem, I found out a pattern. The users' devices who gets this problem right after signIn are not being listed in "Device tracking log" even though their events are shown as "Sign-in passed" in "User event history". Look at this user:
![MicrosoftTeams-image (32)](https://user-images.githubusercontent.com/18200284/158589821-edffcfde-fbd1-4600-a012-50bc78bbb716.png)

If you see, the user tried to log in using two different devices: an iPhone and a SM-G9600 (Android). However, the only device which is listed in "Device tracking log" is the iPhone. It is a pattern, all user who had this problem I can observe the same behavior: I see the sign in event in "User event history" but I don't see the device which I see the logs in "Device tracking log".
.

**Comment by brunovsiqueira** [comment]
Look at a normal user behavior (my user):
![MicrosoftTeams-image (35)](https://user-images.githubusercontent.com/18200284/158654689-2faf1e94-5bbf-4ab0-8fbd-d95b7bb609f4.png)
If you see, every time I make a successful signIn after reinstalling the app, he register my device in the "Device tracking log".

Now look at this user behavior. He said he always uninstall apps for security reasons. He used to be able to sign in our app weeks ago. Now when he signs in after reinstalling the app (in the last attempts), no new device is added to the "Device tracking logs" even though he signed in "successfully" (sign in passes but SessionExpired is thrown when getting credentials). 

![MicrosoftTeams-image (36)](https://user-images.githubusercontent.com/18200284/158655617-d7635e43-ef99-447c-a952-d0dc34abdd69.png)

**Comment by gpanshu** [comment]
Session expired event will be thrown on expired token fetch inside fetchAuthSession after this [PR](https://github.com/aws-amplify/amplify-android/pull/1779) is merged in for Android.

**Comment by Jordan-Nelson** [comment]
Hello everyone - We have been unable to reproduce this issue. However, I am going to label this as a bug since the info provided indicates there may be a race condition present.

We have recently released a new developer preview version of the Auth category ([amplify_flutter v1.0.0-next.0](https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.0)). This new version is a full re-write of the auth category in dart. While the primary purpose of this re-write was to add support for Web and Desktop, we also implemented a new design using state machines that is intended to solve issues such as race conditions. You can read more about the developer preview release in the [blog post](https://aws.amazon.com/blogs/mobile/amplify-flutter-auth-web-and-desktop/) and [docs](https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/).

I believe this will address the issue described above. If you are willing to experiment with this new version while it is in dev-preview we would appreciate any feedback you have.

I will provide an update on this ticket once this version is out of dev-preview and is ready for production use.

Let me know if you have any questions.

**Comment by brunovsiqueira** [comment]
Hi @Jordan-Nelson. Any plans to make the new version of amplify auth stable?

**Comment by petreikis** [comment]
We are also facing the same issue. @brunovsiqueira, did you find any workarounds to mitigate it?

**Comment by brunovsiqueira** [comment]
@petreikis I didn't. I think there is no possible workarounds. Amplify lib is the one that should deal with it.
@Jordan-Nelson Any plans to release the new version?

**Comment by wiktoriasobczyk** [comment]
I'm waiting for the amplify lib update, mby it would be working better in the next version 🤷 😢

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira - I don't have a timeline on when the dev-preview will be considered stable. The latest version is on pub: https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.3

@petreikis @wiktoriasobczyk - Are you able to reproduce this consistently? If so, please do let me know. The first step towards resolving this would be reproducing it consistently. Unfortunately we have not been able to do that yet.

**Comment by brunovsiqueira** [comment]
It is still happening with some users. I can't reproduce but it does not mean that the problem does not exist. I am still waiting for the new lib update, to see if it stops happening.

**Comment by brunovsiqueira** [comment]
Hi @Jordan-Nelson. This issue is still affecting production users. Is there any plans for the mentioned release candidate to become stable? I see you guys are still working on that, but I am a little bit afraid of updating our app to a non stable version.

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira - v1.0.0 of Amplify Flutter is now stable. We believe that this issue has been addressed. Would you please upgrade to the latest version and let us know if you still observe this issue?

This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).

**Comment by brunovsiqueira** [comment]
Thanks a lot @Jordan-Nelson. We are working in the migration and I let you guys know as soon as I have an update.

**Comment by brunovsiqueira** [comment]
Hi @Jordan-Nelson. While we were migrating we realized that the minSdkVersion is 24 and unfortunately we have thousands of users below this SDK version, what makes us not to migrate.

Is there any reason for upgrading the minSdkVersion? Is there any possibility to downgrade?

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira - Correct, the minSdk was bumped from 21 to 24. There were a few reasons for this. What is the min version you currently support? Do you know roughly how many users you have at each version below 24?

**Comment by brunovsiqueira** [comment]
@Jordan-Nelson We support minSdkVersion 21.
In the last 90 days:
Around 3k users in 23
Around 200 users in 22 and 21.

Is there any chance to downgrade the minSdkVersion in this new version? I think the biggest problem is with Android 6 (API 23).

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira - Are you using other categories from Amplify besides Auth (API, Analytics, DataStore, Storage, or Notifications)?

**Comment by brunovsiqueira** [comment]
@Jordan-Nelson only auth at this time.

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira - I will discuss the possibility of supporting 23 with the team and get back to you. Is this your only blocker from migrating to v1 at this time?

**Comment by brunovsiqueira** [comment]
@Jordan-Nelson yes, it is the only blocker for the tests we have made. The ideal would be supporting 21, because cutting some users, even if it is not many, is always not good. Thanks on advance.

**Comment by brunovsiqueira** [comment]
Hi @Jordan-Nelson. Do you have any updates?

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira I don't have an update yet. There  are a couple dependencies that require level 23/24. We are still discussing if we could support lower API levels.

**Comment by Jordan-Nelson** [comment]
Just wanted to let you know that we are still looking into this. I will provide post here when I have an update.

**Comment by brunovsiqueira** [comment]
Hi @Jordan-Nelson. Thanks for the update. Looking forward to hear back from you.

**Comment by GarbielBarbosa** [comment]
I hope this message finds you well. I wanted to address the issue reported by XXXX. I recently updated the library to version 1.10.0, but some users are still experiencing the same problem. Unfortunately, I don't have enough data at the moment to determine if the error rate has increased or decreased. I am actively seeking additional solutions and will continue to closely monitor the situation.

Thank you for your understanding, and any further feedback would be greatly appreciated.

**Comment by GarbielBarbosa** [comment]
I hope this message finds you well. I wanted to address the issue reported by @brunovsiqueira. I recently updated the library to version 1.10.0, but some users are still experiencing the same problem. Unfortunately, I don't have enough data at the moment to determine if the error rate has increased or decreased. I am actively seeking additional solutions and will continue to closely monitor the situation.

Thank you for your understanding, and any further feedback would be greatly appreciated.

**Comment by GarbielBarbosa** [comment]
I hope this message finds you well. I wanted to address the issue reported by @brunovsiqueira . I recently updated the library to version 1.10.0, but some users are still experiencing the same problem. Unfortunately, I don't have enough data at the moment to determine if the error rate has increased or decreased. I am actively seeking additional solutions and will continue to closely monitor the situation.

Thank you for your understanding, and any further feedback would be greatly appreciated.

**Comment by Jordan-Nelson** [comment]
@GarbielBarbosa Just to confirm, did you mean version 1.1.0?

**Comment by GarbielBarbosa** [comment]
@Jordan-Nelson, I apologize for the mistake. I am using version 1.1.0 of the library.
I am unable to reproduce the error on my devices; the errors are only reported by my users. Additionally, I am able to track them through Crashlytics/Firebase.
Below is the structure of my code.

```
Future<void> hasUser() async {
    final ILocalStorage storage = Modular.get();
    final User? user = await storage.getUser();
  
      // verifica se o usuário está logado.
      try {
        await Amplify.Auth.getCurrentUser();
      } catch (e) {
        showAlertErrorLogin(code: '1000');
        await goToLogin();
        return;
      }

      // Retorna os tokens do cognito e os fixa no usuário
      try {
        final session = await Amplify.Auth.getPlugin(
          AmplifyAuthCognito.pluginKey,
        ).fetchAuthSession();
        if (session.isSignedIn) {
          storage.user.acessToken = session.userPoolTokensResult.value.accessToken.toJson();
          storage.user.idToken = session.userPoolTokensResult.value.idToken.toJson();
          storage.user.refreshToken = session.userPoolTokensResult.value.refreshToken;
        }
      } catch (e, stackTrace) {
        await FirebaseCrashlytics.instance.recordError(e, stackTrace, reason: 'Erro ao logar fetchAuthSession');
        AnalyticsFunctions.saveLog('Erro ao logar fetchAuthSession', {'error': e.toString()});
        try {
          final session = await Amplify.Auth.getPlugin(
            AmplifyAuthCognito.pluginKey,
          ).fetchAuthSession();
          if (session.isSignedIn) {
            storage.user.acessToken = session.userPoolTokensResult.value.accessToken.toJson();
            storage.user.idToken = session.userPoolTokensResult.value.idToken.toJson();
            storage.user.refreshToken = session.userPoolTokensResult.value.refreshToken;
          }
        } catch (e) {
          await FirebaseCrashlytics.instance.recordError(e, stackTrace, reason: 'Erro ao logar fetchAuthSession 2');
          AnalyticsFunctions.saveLog('Erro ao logar fetchAuthSession', {'error': e.toString()});
          showAlertErrorLogin(code: '1002');
          await goToLogin();
          return;
        }
      }

      // Caso o id token do usuário ou o email esteja vazio o usuário é deslogado.
      if (storage.user.idToken == null || storage.user.email == null) {
        await Amplify.Auth.signOut(
          options: const SignOutOptions(),
        );
        showAlertErrorLogin(code: '1003');
        await goToLogin();
        return;
      }

      await goToHome();
    }
  }
```

Would this type of StackTrace be useful?

<img width="1145" alt="Captura de Tela 2023-05-27 às 17 19 15" src="https://github.com/aws-amplify/amplify-flutter/assets/30707007/ca7d0927-c3d4-4d44-b545-b204322ced49">

**Comment by Jordan-Nelson** [comment]
@GarbielBarbosa The stack trace indicates the error is coming from the method channel implementation of fetchAuthSession. Version 0.x of Amplify wraps Amplify iOS and Android using method channels. This implementation is no longer present in version 1.x of Amplify is it has been re-written in Dart. Is it possible that the users that are experiencing this are using a version of your app that still uses version 0.x of Amplify?

**Comment by GarbielBarbosa** [comment]
Hi, @Jordan-Nelson! I'd like to confirm if the correct method to obtain the tokens is as follows:

```
  final session = await Amplify.Auth.getPlugin(
      AmplifyAuthCognito.pluginKey,
    ).fetchAuthSession();
    if (session.isSignedIn) {
      storage.user.acessToken = session.userPoolTokensResult.value.accessToken.toJson();
      storage.user.idToken = session.userPoolTokensResult.value.idToken.toJson();
      storage.user.refreshToken = session.userPoolTokensResult.value.refreshToken;
    }
```

Our builds are performed through a pipeline. I truly believe that the version has already been updated with the new library version.

Please review and confirm if my understanding is correct. If not, please clarify the correct method. Thank you in advance for your assistance!

**Comment by Jordan-Nelson** [comment]
@GarbielBarbosa - Yes, fetchAuthSession is the correct way to obtain the tokens.

 It could be that the users experiencing this have a spotty network connection that is causing only some requests to fail. You can handle the network exception case. A network exception indicates that the user's access/id tokes have expired, but the refresh token may still be valid (there is no way to determine the validity of the refresh token while offline). In the Authenticator UI component this case is handled as a valid session while the user is offline. Once they come back online if the refresh token has expired they will then be direct to sign in again. You can see how that is handled below.

https://github.com/aws-amplify/amplify-flutter/blob/7949173db4ce50d75afa18e61f5ded2784899de1/packages/authenticator/amplify_authenticator/lib/src/services/amplify_auth_service.dart#L204-L224

If you would like further assistance can you please open a new issue since it does not appear that the exception you are seeing is that same as what was reported in the original issue.

**Comment by ilyasarafath** [comment]
getting same issue 
for me its getting after signIn accessing storage upload 
SessionExpiredException {
E/flutter (13937):   "message": "The AWS credentials could not be retrieved",
E/flutter (13937):   "recoverySuggestion": "Invoke Amplify.Auth.signIn to re-authenticate the user",
E/flutter (13937):   "underlyingException": "NotAuthorizedException {\n  message=Token is not from a supported provider of this identity pool.,\n}"
E/flutter (13937): }

**Comment by Jordan-Nelson** [comment]
@ilyasarafath the underlying exception in the logs you shared is different than what has been shared for this issue. I think it is likely a different issue. Can you open a new issue so that we can gather all the details about your issue?

**Comment by brunovsiqueira** [comment]
@Jordan-Nelson have you guys discussed about downgrading the minSdkVersion? We are needing to migrate to see if the issue persists, but we can't without supporting all users.

**Comment by Jordan-Nelson** [comment]
Hi @brunovsiqueira - Apologies that this has been taking a while. There are two blockers from supporting these API levels that we are discussing internally:
1. EncryptedSharedPreferences, used within Auth (as well as other categories), has stable support for 23+
2. Amplify Android (used within a couple non-auth categories) supports 24+

It is unlikely that we will be able to support API 21/22 because of the blocker on EncryptedSharedPreferences. We are still discussing support for 23.

If we were able to support 23, would this allow you to adopt Amplify flutter v1 or would lack of support for 21/22 still be a blocker?

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira - I just wanted to follow up with you and see if supporting API level 23 would unblock you from upgrading. As I had mentioned, I don't think supporting levels 21 & 22 will be possible because of the blocker on EncryptedSharedPreferences.

**Comment by brunovsiqueira** [comment]
Hi @Jordan-Nelson. Thank you for the follow-up.

Unfortunately, we cannot upgrade the minSdkVersion to 23. We currently have 2,000 active users on API 21 and 23, and for now we cannot stop supporting them.

**Comment by Jordan-Nelson** [comment]
> Hi @Jordan-Nelson. Thank you for the follow-up.
> 
> 
> 
> Unfortunately, we cannot upgrade the minSdkVersion to 23. We currently have 2,000 active users on API 21 and 23, and for now we cannot stop supporting them.

Just to confirm, do you mean API 21 and 22?

**Comment by brunovsiqueira** [comment]
> > Hi @Jordan-Nelson. Thank you for the follow-up.
> > Unfortunately, we cannot upgrade the minSdkVersion to 23. We currently have 2,000 active users on API 21 and 23, and for now we cannot stop supporting them.
> 
> Just to confirm, do you mean API 21 and 22?

Exactly. Sorry for the confusion. I have just updated my comment.

**Comment by Jordan-Nelson** [comment]
@brunovsiqueira - I am going to close out this issue. Please feel free to open a request for supporting a lower API level on Android in v1. Please let me know if you have any questions.

**Comment by brunovsiqueira** [comment]
Hi @Jordan-Nelson, how are you?

Our app is now being migrated to support API >= 23 (minSdkVersion 23). Is there a possibility to have Amplify Flutter supporting 23? If so, we can migrate to 1.x.x. That would be really helpful four us at BTG Pactual.

**Comment by Jordan-Nelson** [comment]
Hi @brunovsiqueira - You are look for Amplify V1 to support Android API level 23+ for the Auth category only, correct? You had previously asked about API levels 21/22. Are you only looking for API level 23+?

Could you open a new feature request for this, and include those details? Thanks

**Comment by brunovsiqueira** [comment]
> Hi @brunovsiqueira - You are look for Amplify V1 to support Android API level 23+ for the Auth category only, correct? You had previously asked about API levels 21/22. Are you only looking for API level 23+?
> 
> Could you open a new feature request for this, and include those details? Thanks

Thanks for the fast answer. Yes, I am looking for amplify_flutter, amplify_auth_cognito and amplify_core to support API 23. These are the dependencies we have.

I am going to open a feature request.

**Comment by brunovsiqueira** [comment]
@Jordan-Nelson I have just created the feature request: https://github.com/aws-amplify/amplify-flutter/issues/3739.
